From 515a963c8063621d55574b00b1d5040e3ba2ab9e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hauke=20Kr=C3=BCger?= <hk@binauricsaudio.com>
Date: Mon, 2 Sep 2024 18:04:16 +0200
Subject: [PATCH] Modified lib to not exit in case a bmp was not valid

---
 cbmp.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/cbmp.c b/cbmp.c
index 0cfd4f0..85d5a8d 100644
--- a/cbmp.c
+++ b/cbmp.c
@@ -51,7 +51,7 @@ BMP* bopen(char* file_path)
     if (fp == NULL)
     {
         perror("Error opening file");
-        exit(EXIT_FAILURE);
+        return NULL;
     }
 
     BMP* bmp = (BMP*) malloc(sizeof(BMP));
@@ -62,6 +62,7 @@ BMP* bopen(char* file_path)
     if(!_validate_file_type(bmp->file_byte_contents))
     {
         _throw_error("Invalid file type");
+        return NULL;
     }
 
     bmp->pixel_array_start = _get_pixel_array_start(bmp->file_byte_contents);
@@ -73,6 +74,7 @@ BMP* bopen(char* file_path)
     if(!_validate_depth(bmp->depth))
     {
         _throw_error("Invalid file depth");
+        return NULL;
     }
 
     _populate_pixel_array(bmp);
@@ -172,7 +174,6 @@ void bclose(BMP* bmp)
 void _throw_error(char* message)
 {
     fprintf(stderr, "%s\n", message);
-    exit(1);
 }
 
 unsigned int _get_int_from_buffer(unsigned int bytes, 
@@ -208,6 +209,7 @@ unsigned char* _get_file_byte_contents(FILE* fp, unsigned int file_byte_number)
     if (result != file_byte_number)
     {
         _throw_error("There was a problem reading the file");
+        return NULL;
     }
 
 
-- 
2.35.1.windows.2

