--- ffmpeg/libavcodec/x86/v210-init.c	2023-03-01 16:25:04.167064800 +0100
+++ ffmpeg.orig/libavcodec/x86/v210-init.c	2023-03-01 19:01:47.385426800 +0100
@@ -16,7 +16,6 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
-#include <assert.h>
 #include "libavutil/attributes.h"
 #include "libavutil/x86/cpu.h"
 #include "libavcodec/v210dec.h"
@@ -47,10 +46,7 @@
             s->unpack_frame = ff_v210_planar_unpack_aligned_avx2;
 
         if (EXTERNAL_AVX512ICL(cpu_flags))
-		{
-			assert(0);
-            // s->unpack_frame = ff_v210_planar_unpack_avx512icl;
-		}
+            s->unpack_frame = ff_v210_planar_unpack_avx512icl;
     }
     else {
         if (cpu_flags & AV_CPU_FLAG_SSSE3)
@@ -63,10 +59,7 @@
             s->unpack_frame = ff_v210_planar_unpack_unaligned_avx2;
 
         if (EXTERNAL_AVX512ICL(cpu_flags))
-		{
-			assert(0);
-            // s->unpack_frame = ff_v210_planar_unpack_avx512icl;
-		}
+            s->unpack_frame = ff_v210_planar_unpack_avx512icl;
     }
 #endif
 }
