diff -ur civetweb.orig/src/civetweb.c civetweb/src/civetweb.c
--- civetweb.orig/src/civetweb.c	2023-01-29 13:00:05.381473100 +0100
+++ civetweb/src/civetweb.c	2023-01-29 12:57:22.147221400 +0100
@@ -1123,7 +1123,7 @@
 #if defined(_WIN64) && !defined(NO_ATOMICS)
 	ret = InterlockedIncrement64(addr);
 #elif defined(_WIN32) && !defined(NO_ATOMICS)
-	ret = InterlockedIncrement(addr);
+	ret = InterlockedIncrement((LONG*)addr);
 #elif defined(__GNUC__)                                                        \
     && ((__GNUC__ > 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ > 0)))           \
     && !defined(NO_ATOMICS)
@@ -1146,7 +1146,7 @@
 #if defined(_WIN64) && !defined(NO_ATOMICS)
 	ret = InterlockedDecrement64(addr);
 #elif defined(_WIN32) && !defined(NO_ATOMICS)
-	ret = InterlockedDecrement(addr);
+	ret = InterlockedDecrement((LONG*)addr);
 #elif defined(__GNUC__)                                                        \
     && ((__GNUC__ > 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ > 0)))           \
     && !defined(NO_ATOMICS)
@@ -11967,7 +11967,7 @@
 			    get_rel_url_at_current_server(destination_hdr, conn);
 			if (h) {
 				int len = (int)strlen(h);
-				local_dest = mg_malloc_ctx(len + 1, conn->phys_ctx);
+				local_dest = (char*)mg_malloc_ctx(len + 1, conn->phys_ctx);
 				mg_url_decode(h, len, local_dest, len + 1, 0);
 			}
 		}
@@ -12611,7 +12611,7 @@
 	}
 
 	link_concat_len = strlen(uri) + strlen(name) + 1;
-	link_concat = mg_malloc_ctx(link_concat_len, conn->phys_ctx);
+	link_concat = (char*)mg_malloc_ctx(link_concat_len, conn->phys_ctx);
 	if (!link_concat) {
 		return 0;
 	}
@@ -22353,7 +22353,7 @@
 				len += 2;
 			}
 		}
-		all_methods = mg_malloc(len);
+		all_methods = (char*)mg_malloc(len);
 		if (!all_methods) {
 			/* Must never happen */
 			mg_global_unlock();
